executor/index_lookup_hash_join.go

```diff
@@ -452,13 +453,14 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 		if !ok {
 			break
 		}
+		// We need to init resultCh before the err is returned.
+		if task.keepOuterOrder {
+			resultCh = task.resultCh
+		}
```

Comment:
Please cover this path in the unit test. The `testIndexHashJoinInnerWorkerErr` failpoint can be used.